### PR TITLE
Submission ui alignment

### PR DESCRIPTION
**Pull request by kolyaopahle**
## Checklist

* [x] Test your changes as thoroughly as possible before you commit them. Preferably, automate your test by unit/integration tests.
* [ ] If this PR comes from a fork, please [Allow edits from maintainers](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork)
* [x] Set a speaking title. Format: {task_name} (closes #{issue_number}). For example: Use logger (closes #41)
* [ ] [Link your Pull Request to an issue](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) (if applicable)
* [ ] Create Work In Progress [WIP] pull requests only if you need clarification or an explicit review before you can continue your work item.
* [x] Make sure that your PR is not introducing _unncessary_ reformatting (e.g., introduced by on-save hooks in your IDE)
* [x] Check our [Contribution Guidelines](https://github.com/corona-warn-app/cwa-app-android/blob/master/CONTRIBUTING.md)

## Description
This PR combines lots of new submission flow ui changes and fragments into one cohesive submission experience.

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/SubmissionDispatcherFragment.kt [hunk 1]
```diff
@@ -48,4 +51,77 @@ class SubmissionDispatcherFragment : BaseFragment() {
...
+                    REQUEST_CAMERA_PERMISSION_CODE
+                )
+            }
+        } else {
+            cameraPermissionIsGranted()
+        }
+    }
+
+    private fun showCameraPermissionRationaleDialog() {
+        val alertDialog: AlertDialog = requireActivity().let {
```
Comment (harambasicluka) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/SubmissionDispatcherFragment.kt:71]: We added a DialogHelper, see OnboardingNotificationsFragment.kt.

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/SubmissionDispatcherFragment.kt [hunk 2]
```diff
@@ -48,4 +51,77 @@ class SubmissionDispatcherFragment : BaseFragment() {
...
+        super.onRequestPermissionsResult(requestCode, permissions, grantResults)
+        if (grantResults.isNotEmpty() && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+            cameraPermissionIsGranted()
+        } else {
+            showCameraPermissionDeniedDialog()
+        }
+    }
+
+    private fun showCameraPermissionDeniedDialog() {
+        val alertDialog: AlertDialog = requireActivity().let {
```
Comment (harambasicluka) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/SubmissionDispatcherFragment.kt:107]: Same here, please use the DialogHelper.

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/SubmissionQRCodeScanFragment.kt [hunk 3]
```diff
@@ -134,84 +127,10 @@ class SubmissionQRCodeScanFragment : BaseFragment() {
         alertDialog.show()
```
Comment (harambasicluka) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/SubmissionQRCodeScanFragment.kt:127]: Please also exchange for the new DialogHelper.

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/SubmissionResultPositiveOtherWarningFragment.kt [hunk 4]
```diff
@@ -53,14 +74,11 @@ class SubmissionResultPositiveOtherWarningFragment : BaseFragment() {
                 setPositiveButton(
```
Comment (harambasicluka) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/SubmissionResultPositiveOtherWarningFragment.kt:74]: Can also be replaced by the DialogHelper.

File: Corona-Warn-App/src/main/res/layout/include_test_result_card.xml [hunk 5]
```diff
@@ -31,27 +29,26 @@
...
-                android:layout_height="100dp"
-                android:background="@{FormatterSubmissionHelper.formatTestStatusIcon(testResult)}"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
                 app:layout_constraintBottom_toBottomOf="parent"
+                android:background="@{FormatterSubmissionHelper.formatTestStatusIcon(testResult)}"
                 app:layout_constraintEnd_toEndOf="parent" />
 
             <TextView
                 android:id="@+id/test_result_card_headline"
```
Comment (harambasicluka) [inline Corona-Warn-App/src/main/res/layout/include_test_result_card.xml:39]: If it's a headline it should use the style `textMultiline`.

**Review by harambasicluka — CHANGES_REQUESTED** [review]
Nice PR :) 
My main focus is on the DialogHelpers we introduced last night. So please use them to be consistent.

File: Corona-Warn-App/src/main/res/layout/include_test_result_card.xml (hunk 5 above)
Comment (kolyaopahle) [reply Corona-Warn-App/src/main/res/layout/include_test_result_card.xml:39]: it does, attribute sorting just moved it down a few lines

**Comment by jakobmoellersap** [comment]
can you update dev and retry again?

**Review by jakobmoellerdev — APPROVED** [review]

**Pull request merged**